### PR TITLE
REF: ImportDiscoveryWallet to TSX

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
	- Enhanced wallet import functionality with additional parameters for better user input.
	- Improved user interface for wallet discovery, including new components for empty and header states.

- **Bug Fixes**
	- Resolved import path case sensitivity issues to ensure proper component loading.

- **Documentation**
	- Updated type definitions for better clarity and type safety within the wallet import component.

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: screen/wallets/ImportWalletDiscovery.tsx [hunk 1]
```diff
@@ -99,16 +126,22 @@ const ImportWalletDiscovery = () => {
         IdleTimerManager.setIdleTimerDisabled(false);
       });
 
-    return () => task.current.stop();
+    return () => {
+      if (task.current) {
+        task.current.stop();
+      }
```
Comment (Overtorment) [inline screen/wallets/ImportWalletDiscovery.tsx:132]: can be shortened to `task.current?.stop?.();`

File: screen/wallets/ImportWalletDiscovery.tsx [hunk 2]
```diff
@@ -99,16 +126,22 @@ const ImportWalletDiscovery = () => {
...
+    };
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   const handleCustomDerivation = () => {
-    task.current.stop();
+    if (task.current) {
+      task.current.stop();
+    }
```
Comment (Overtorment) [inline screen/wallets/ImportWalletDiscovery.tsx:140]: can be shortened

**Review by Overtorment — APPROVED** [review]
mosstly lgtm, some minor issues.

@limpbrains your faivourite feature affected, you might wana take a look pls

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**